### ISSUE TITLE
AdminAuth using token from Cloudflare fails to assign user

### ISSUE THREAD

**Issue by notscottsmith** [node-red/node-red#3341]
### Current Behavior

Authentication fails. A screen showing login dialog (without any input field) appears and nothing else. Browser console log indicates an unauthorized response from Node-RED

### Expected Behavior

User should be authenticated

### Steps To Reproduce

Configure header based authentication for Node-RED

```
adminAuth: {
            tokens: function(token) {
                return new Promise(function(resolve, reject) {
                    var jwt = require('jsonwebtoken');
                    var jwksClient = require('jwks-rsa');
                
                    var client = jwksClient({
                        jwksUri: process.env.NODE_RED_JWKS_URI,
                    });
                
                    var options = {
                        algorithms: [ 'RS256' ],
                        audience: process.env.NODE_RED_JWT_AUDIENCE,
                        issuer: process.env.NODE_RED_JWT_ISSUER
                    };
                
                    function getKey(header, callback){
                        client.getSigningKey(header.kid, function(err, key) {
                            var signingKey = null;
                            try {
                                signingKey = key.publicKey || key.rsaPublicKey;
                            } catch (e) {
                                console.log(e);
                            }
                            callback(null, signingKey);
                        });
                    }
                
                    jwt.verify(token, getKey, options, function(err, decoded) {
                        if (err) {
                            console.log("Error authenticating: " + err);
                            resolve(null);
                        } else {
                            var user = { username: decoded.email, permissions: "*" };
                            
                            resolve(user);
                        }
                    });
                });
            },
            tokenHeader: 'cf-access-jwt-assertion'
        }
```

### Environment

- Node-RED version: 2.1.4
- Node.js version: 14.18.2
- npm version: 6.14.15
- Platform/OS: debian latest
- Browser: chrome/firefox

Node-RED is running in a container, compiled with the jwt-rsa and javawebtoken modules installed. Previous version deployed, v1.3.5, authenticates the user successfully with the identical adminAuth section.

Currently, "user" variable is successfully being created and applied with data decoded from the jwt but the user is still not being authenticated.

Please also see: https://discourse.nodered.org/t/adminauth-tokens-function-for-jwt-works-in-v1-3-5-but-not-in-v2-1-4

**Comment by knolleary** [comment]
I can recreate this.

For ease of testing, you can use `adminAuth` as follows:

```
adminAuth: {
         tokens: async function(token) {
             return {username: "nick@flowforge.com", permissions:"*"}
         },
         tokenHeader: 'host'
     },
```

By using `host` as `tokenHeader` there's no need to setup any special environment to inject a header (all requests have a `host` header.

With this config, I can access a 1.3.x instance, but not a 2.x one.

Now to figure out what we broke...

**Comment by knolleary** [comment]
Tracing it through, I can see we're validating the request and telling passport it is a valid request:

https://github.com/node-red/node-red/blob/f8c47f59bce1db26d297751b459e03a38881cfca/packages/node_modules/%40node-red/editor-api/lib/auth/strategies.js#L164

For some reason, Passport disagrees. Going to have to put on a hard hat and start digging into Passport internals. Not my favourite activity.

**Comment by knolleary** [comment]
Managed to trace it through passport and back out into our code.

Fixed via #3343  - will be in 2.1.5.
